src/api/block_builder/input_selection/sender_issuer.rs

```diff
@@ -252,14 +199,190 @@ impl<'a> ClientBlockBuilder<'a> {
             }
         }
 
-        // Check required alias and nft outputs with new added outputs
-        // no need to check for sender and issuer again, since these outputs already exist and we don't set new features
-        // for them
+        // Check required Alias and NFT outputs with new added outputs.
+        // No need to check for sender and issuer again, since these outputs already exist and we don't set new features
+        // for them.
         let utxo_chain_inputs = self
             .get_utxo_chains_inputs(required_inputs.iter().map(|i| &i.output))
             .await?;
         required_inputs.extend(utxo_chain_inputs.into_iter());
 
-        Ok((force_use_all_inputs, required_inputs))
+        Ok(required_inputs)
+    }
+}
+
+// Select inputs for sender and issuer features
+pub(crate) fn select_inputs_for_sender_and_issuer<'a>(
+    inputs: impl Iterator<Item = &'a InputSigningData> + Clone,
+    selected_inputs: &mut Vec<InputSigningData>,
+    selected_inputs_output_ids: &mut HashSet<OutputId>,
+    outputs: &mut Vec<Output>,
+    current_time: u32,
+) -> Result<()> {
+    log::debug!("[select_inputs_for_sender_and_issuer]");
+
+    let all_required_addresses = get_required_addresses_for_sender_and_issuer(selected_inputs, outputs, current_time)?;
+    'addresses_loop: for address in all_required_addresses {
+        match address {
+            Address::Ed25519(_) => {
+                // first check already selected outputs
+                for input_signing_data in selected_inputs.iter() {
+                    if output_contains_address(
+                        &input_signing_data.output,
+                        input_signing_data.output_id()?,
+                        &address,
+                        current_time,
+                    ) {
+                        continue 'addresses_loop;
+                    }
+                }
+
+                // if not found, check currently not selected outputs
+                for input_signing_data in inputs.clone() {
+                    // Skip already added inputs
+                    let output_id = input_signing_data.output_id()?;
+                    if selected_inputs_output_ids.contains(&output_id) {
+                        continue;
+                    }
+                    if output_contains_address(&input_signing_data.output, output_id, &address, current_time) {
+                        selected_inputs.push(input_signing_data.clone());
+                        selected_inputs_output_ids.insert(output_id);
+                        // break when we added the required output to the selected_inputs
+                        continue 'addresses_loop;
+                    }
+                }
+
+                return Err(Error::MissingInput(format!(
```

Comment:
Use MissingInputWithEd25519Address instead?

Reply:
Hmm but it would be a bit inconsistent with the 2 other 🤔

Reply:
I also changed it because we could have an input with an Ed25519 address, but not the correct one
You wouldn't know that it's missing for sender or issuer with `MissingInputWithEd25519Address`